src/traces/candlestick/attributes.js

```diff
@@ -0,0 +1,42 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var OHLCattrs = require('../ohlc/attributes');
+var boxAttrs = require('../box/attributes');
+
+var directionAttrs = {
+    name: OHLCattrs.increasing.name,
+    showlegend: OHLCattrs.increasing.showlegend,
+
+    color: Lib.extendFlat({}, boxAttrs.line.color),
+    width: Lib.extendFlat({}, boxAttrs.line.width),
```

Comment:
I probably should have noticed this earlier (sorry!) but `color` and `width` seem odd for line color and width. It took me a bit to figure out that this wasn't the width of the candle, and that we don't *need* an attribute for that because `layout.boxgap` does that... This should probably be mentioned somewhere!

Also just like `whiskerwidth` I'd think `width` is a property shared by both directions. Can we call these `linecolor` (in `directionAttrs`) and `linewidth` (in `attributes`)? I guess we could argue that for consistency these should go inside a `line` container instead... it seems a bit funny to make a one-member container just to keep the names consistent but maybe that will actually be easier and clearer.